api/catalog/api/serializers/media_serializers.py

```diff
@@ -149,6 +156,14 @@ def validate_tags(self, value):
     def validate_title(self, value):
         return self._truncate(value)
 
+    def validate_page_size(self, value):
+        request = self.context.get("request", None)
+        if request.user and request.user.is_anonymous and value > 20:
+            raise get_api_exception(
+                "Page size must be between 1 & 20 for unauthenticated requests.", 401
+            )
+        return value
```

Comment:
The `None` default is redundant on dictionary's `get` method, but more importantly we need to actually handle the None case here or we'll end up with 500s:

```suggestion
        request = self.context.get("request", None)
        authenticated = request and request.user and not request.user.is_anonymous
        if not authenticated and value > 20:
            raise get_api_exception(
                "Page size must be between 1 & 20 for unauthenticated requests.", 401
            )
        return value
```

That seems like a reasonable and safe way to handle None. What you do think? Just treat it the same as an anonymous request?

Reply:
I don't think that case will happen but I like this! Though it's a bit tricky to check using the condition for the authenticated user (it becomes `None` for some reason 🤷🏻‍♀️). I modified it a bit, LMKWYT.

Reply:
Looks good to me! I also don't foresee it happening but "something something [Murphy's Law](https://en.wikipedia.org/wiki/Murphy%27s_law)" :sweat_smile: